cpp/unreal_plugins/SimulationController/Source/SimulationController/SimulationController.cpp

```diff
@@ -36,27 +36,26 @@
 #include "SimulationController/RpcServer.h"
 #include "SimulationController/SphereAgent.h"
 #include "SimulationController/Task.h"
+#include "SimulationController/UrdfBotAgent.h"
 #include "SimulationController/Visualizer.h"
 
 // Different possible frame states for thread synchronization
-enum class FrameState
-{
-    Idle,
-    RequestPreTick,
-    ExecutingPreTick,
-    ExecutingTick,
-    ExecutingPostTick
-};
+enum class FrameState { Idle, RequestPreTick, ExecutingPreTick, ExecutingTick, ExecutingPostTick };
 
 void SimulationController::StartupModule()
 {
     std::cout << "[SPEAR | SimulationController.cpp] SimulationController::StartupModule" << std::endl;
 
     ASSERT(FModuleManager::Get().IsModuleLoaded(TEXT("CoreUtils")));
     ASSERT(FModuleManager::Get().IsModuleLoaded(TEXT("OpenBot")));
-    
-    post_world_initialization_delegate_handle_ = FWorldDelegates::OnPostWorldInitialization.AddRaw(
-        this, &SimulationController::postWorldInitializationEventHandler);
+    ASSERT(FModuleManager::Get().IsModuleLoaded(TEXT("UrdfBot")));
+
+    if (!Config::s_initialized_) {
```

Comment:
Need to add this check to ShutdownModule() so we're doing symmetric initialization/termination work.

Reply:
fixed